integration_tests/structs_29.py

```diff
@@ -0,0 +1,20 @@
+from lpython import dataclass, i32
+
+@dataclass
+class Foo:
+     x: i32
+     y: i32
+
+
+def main0() -> None:
+    x: list[Foo]
+    y: Foo = Foo(0, 1)
+    z: Foo = Foo(1, 2)
+    x = [y, z]
+    i: i32 = 0
+    for y in x:
+        assert y.x == i
+        assert y.y == i+1
+        i += 1
```

Comment:
> Does a copy happen here --- so y in the loop is a copy of either the y or z variables? So if you modify y in the loop, it will not reflect in the original structure?

Yes, it affects the original one too. CPython does the same. See the below output on the following snippet:
```py
from lpython import dataclass, i32

@dataclass
class Foo:
     x: i32
     y: i32


def main0() -> None:
    x: list[Foo]
    y: Foo = Foo(0, 1)
    z: Foo = Foo(1, 2)
    x = [y, z]
    i: i32 = 0
    for y in x:
        assert y.x == i
        assert y.y == i+1
        i += 1
    print(y.x, y.y)

main0()
```

Outputs:
```
% lpython integration_tests/structs_29.py 
1 2
% PYTHONPATH=$PWD/src/runtime/lpython python integration_tests/structs_29.py
1 2

```

Reply:
> So if you modify y in the loop, it will not reflect in the original structure?

Yeah, you are right.

```py
from lpython import dataclass, i32

@dataclass
class Foo:
     x: i32
     y: i32


def main0() -> None:
    x: list[Foo]
    y: Foo = Foo(0, 1)
    z: Foo = Foo(1, 2)
    x = [y, z]
    i: i32 = 0
    for y in x:
        assert y.x == i
        assert y.y == i+1
        y.x = 10
        y.y = i + y.x
        i += 1
    y = x[0]
    print(y.x, y.y)
    y = x[1]
    print(y.x, y.y)

main0()
```
This gives:

```
% PYTHONPATH=$PWD/src/runtime/lpython python c.py
10 10
10 11
% lpython c.py                                   
0 1
1 2
```

Reply:
Right. So in CPython, the inner loop `y` is a "pointer", while in LPython it is a copy.

This is a bigger issue to design and fix: https://github.com/lcompilers/lpython/issues/1746#issuecomment-1520940634. 

For now it's fine, but we'll have to tackle this soon.